auth-foundation/src/test/java/com/okta/authfoundation/client/NetworkUtilsTest.kt

```diff
@@ -851,6 +852,51 @@ class NetworkingTest {
         assertThat(events.last().retryCount).isEqualTo(10)
     }
 
+    @Test fun testPerformRequestRetryClosesUnusedResponsesAfterEmittingEvent(): Unit = runBlocking {
+        val events = mutableListOf<RateLimitExceededEvent>()
+        val rateLimitEventHandler = object : EventHandler {
+            override fun onEvent(event: Any) {
+                val errorEvent = event as RateLimitExceededEvent
+                errorEvent.minDelaySeconds = 0L
+                events.add(errorEvent)
+                assertThat(event.response.peekBody(Long.MAX_VALUE).string()).isEqualTo("")
+            }
+        }
+        oktaRule.eventHandler.registerEventHandler(rateLimitEventHandler)
+
+        for (i in 0 until 4) { // One try + 3 retries
+            oktaRule.enqueue(path("/test")) { response ->
+                response.setResponseCode(429)
+                setRateLimitHeaders(response, rateLimitResetEpochTime = TIME_EPOCH_SECS - 500)
+            }
+        }
+
+        val request = Request.Builder()
+            .url(oktaRule.baseUrl.newBuilder().addPathSegments("test").build())
+            .build()
+
+        val response = oktaRule.createOidcClient().performRequest(request)
+
+        var closedResponses = 0
+        assertThat(events.size).isEqualTo(4)
+        val lastRetryEvent = events.removeLast()
+        events.forEach { event ->
+            try {
```

Comment:
This could be rewritten with `assertFailsWith`, but if not, we'll want to add a `fail` in the catch block as well (if the message isn't closed).

It's not super critical since you're asserting the failed count below. But it's a best practice to fail (which is why the `assertFailsWith` exists!